pkg/skaffold/gcbreposv2/repo_resolver.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2024 The Skaffold Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gcbreposv2
+
+import (
+	"context"
+	"fmt"
+
+	cloudbuild "cloud.google.com/go/cloudbuild/apiv2"
+	cloudbuildpb "cloud.google.com/go/cloudbuild/apiv2/cloudbuildpb"
+	"github.com/googleapis/gax-go/v2"
+)
+
+type cloudBuildRepoClient interface {
+	GetRepository(ctx context.Context, req *cloudbuildpb.GetRepositoryRequest, opts ...gax.CallOption) (*cloudbuildpb.Repository, error)
+	FetchReadToken(ctx context.Context, req *cloudbuildpb.FetchReadTokenRequest, opts ...gax.CallOption) (*cloudbuildpb.FetchReadTokenResponse, error)
+	Close() error
+}
+
+var RepositoryManagerClient = repositoryManagerClient
+
+func GetRepoInfo(ctx context.Context, gcpProject, gcpRegion, gcpConnectionName, gcpRepoName string) (repoURI string, readAccessToken string, err error) {
```

Comment:
can we use a struct for (repoURI string, readAccessToken string) , or we can also construct the uri with token here 

https://oauth2:<token>@<repo>

Reply:
Got it! I changed it to return a struct as you suggested. The struct contains two fields, the `CloneRepoURI` with the oauth2 format, and the `URI` without the token. The reason to return both is to use the CloneRepoURI to execute all the needed git commands (clone, fetch, etc), and the URI for the errors we print so we don't show the token in the execution logs. The idea is to re use the logic that we already have in [pkg/skaffold/git/gitutil.go](https://github.com/GoogleContainerTools/skaffold/blob/e4f19756b95020bff442938044a4a03adc01a702/pkg/skaffold/git/gitutil.go#L92) to trigger the repo download/sync. Let me know what do you think.